From 8f0c526232127ec414aedd723bdbf4954c6af0ed Mon Sep 17 00:00:00 2001
From: Srinivas Kandagatla <srinivas.kandagatla@linaro.org>
Date: Wed, 9 May 2018 13:56:12 +0100
Subject: [PATCH 267/334] soc: qcom dt-bindings: Add APR bus bindings

This patch add dt bindings for Qualcomm APR (Asynchronous Packet Router)
bus driver. This bus is used for communicating with DSP which provides
audio and various other services to cpu.

Signed-off-by: Srinivas Kandagatla <srinivas.kandagatla@linaro.org>
Reviewed-by: Rob Herring <robh@kernel.org>
Reviewed-by: Bjorn Andersson <bjorn.andersson@linaro.org>
Reviewed-by: Banajit Goswami <bgoswami@codeaurora.org>
Signed-off-by: Mark Brown <broonie@kernel.org>

Conflicts:
	Documentation/devicetree/bindings/soc/qcom/qcom,apr.txt
	include/dt-bindings/soc/qcom,apr.h
---
 .../devicetree/bindings/soc/qcom/qcom,apr.txt | 49 +++++++++----------
 include/dt-bindings/soc/qcom,apr.h            |  1 +
 2 files changed, 25 insertions(+), 25 deletions(-)

diff --git a/Documentation/devicetree/bindings/soc/qcom/qcom,apr.txt b/Documentation/devicetree/bindings/soc/qcom/qcom,apr.txt
index 6071a9038b4b..bcc612cc7423 100644
--- a/Documentation/devicetree/bindings/soc/qcom/qcom,apr.txt
+++ b/Documentation/devicetree/bindings/soc/qcom/qcom,apr.txt
@@ -9,9 +9,9 @@ used for audio/voice services on the QDSP.
 	Value type: <stringlist>
 	Definition: must be "qcom,apr-v<VERSION-NUMBER>", example "qcom,apr-v2"
 
-- qcom,apr-dest-domain-id
+- reg
 	Usage: required
-	Value type: <prop-encoded-array>
+	Value type: <u32>
 	Definition: Destination processor ID.
 	Possible values are :
 			1 - APR simulator
@@ -28,10 +28,10 @@ of the nodes are not important. The properties of these nodes are defined
 by the individual bindings for the specific service
 - All APR services MUST contain the following property:
 
-- qcom,apr-svc-id
+- reg
 	Usage: required
-	Value type: <prop-encoded-array>
-	Definition: APR Service ID, used for matching the service.
+	Value type: <u32>
+	Definition: APR Service ID
 	Possible values are :
 			3 - DSP Core Service
 			4 - Audio Front End Service.
@@ -45,41 +45,40 @@ by the individual bindings for the specific service
 			12 - Ultrasound stream manager.
 			13 - Listen stream manager.
 
-- qcom,apr-svc-name
-	Usage: required
-	Value type: <stringlist>
-	Definition: User readable name of a APR service.
-
 = EXAMPLE
 The following example represents a QDSP based sound card on a MSM8996 device
 which uses apr as communication between Apps and QDSP.
 
-	apr {
+	apr@4 {
 		compatible = "qcom,apr-v2";
-		qcom,smd-channels = "apr_audio_svc";
-		qcom,apr-dest-domain-id = <APR_DOMAIN_ADSP>;
+		reg = <APR_DOMAIN_ADSP>;
 
-		q6core {
+		q6core@3 {
 			compatible = "qcom,q6core";
-			qcom,apr-svc-name = "CORE";
-			qcom,apr-svc-id = <APR_SVC_ADSP_CORE>;
+			reg = <APR_SVC_ADSP_CORE>;
 		};
 
-		q6afe {
+		q6afe@4 {
 			compatible = "qcom,q6afe";
-			qcom,apr-svc-name = "AFE";
-			qcom,apr-svc-id = <APR_SVC_AFE>;
+			reg = <APR_SVC_AFE>;
+
+			dais {
+				#sound-dai-cells = <1>;
+				hdmi@1 {
+					reg = <1>;
+				};
+			};
 		};
 
-		q6asm {
+		q6asm@7 {
 			compatible = "qcom,q6asm";
-			qcom,apr-svc-name = "ASM";
-			qcom,apr-svc-id = <APR_SVC_ASM>;
+			reg = <APR_SVC_ASM>;
+			...
 		};
 
-		q6adm {
+		q6adm@8 {
 			compatible = "qcom,q6adm";
-			qcom,apr-svc-name = "ADM";
-			qcom,apr-svc-id = <APR_SVC_ADM>;
+			reg = <APR_SVC_ADM>;
+			...
 		};
 	};
diff --git a/include/dt-bindings/soc/qcom,apr.h b/include/dt-bindings/soc/qcom,apr.h
index 905503f81885..006362400c0f 100644
--- a/include/dt-bindings/soc/qcom,apr.h
+++ b/include/dt-bindings/soc/qcom,apr.h
@@ -1,3 +1,4 @@
+/* SPDX-License-Identifier: GPL-2.0 */
 #ifndef __DT_BINDINGS_QCOM_APR_H
 #define __DT_BINDINGS_QCOM_APR_H
 
-- 
2.24.1

